src/cliffordsynthesis/encoding/SingleGateEncoder.cpp

```diff
@@ -119,6 +155,33 @@ LogicTerm SingleGateEncoder::createNoChange(
   return changes;
 }
 
+LogicTerm SingleGateEncoder::createNoChangeOnQubit(const std::size_t pos,
+                                                   const std::size_t q) {
+  auto noChange = LogicTerm(true);
+  noChange      = noChange && (tvars->x[pos + 1][q] == tvars->x[pos][q]);
+  noChange      = noChange && (tvars->z[pos + 1][q] == tvars->z[pos][q]);
+  return noChange;
+}
+
+LogicTerm SingleGateEncoder::createNoGateOnQubit(const std::size_t pos,
+                                                 const std::size_t q) {
+  const auto& singleQubitGates = vars.gS[pos];
+  auto        noGate           = LogicTerm(true);
+  for (std::size_t i = 1; i < SINGLE_QUBIT_GATES.size(); ++i) {
+    noGate = noGate && !singleQubitGates[i][q];
+  }
```

Comment:
This implicitly assumes that the `None` gate is at position 0. While this is a fair assumption, I believe it would be cleaner to 
 - either iterate over the SINGLE_QUBIT_GATES and skip over the None entry
 - get the index of the None entry before the loop and explicitly skip over its index in the loop.

Reply:
Since the `None` entry is technically considered a gate one can actually include it in the loop anyway. But a `None` gate also doesn't hurt, so I will do it the first way you proposed.